scripts/deno/surface-lighthouse-results.ts

```diff
@@ -0,0 +1,206 @@
+import { Octokit } from "https://cdn.skypack.dev/octokit?dts";
+import type { RestEndpointMethodTypes } from "https://cdn.skypack.dev/@octokit/plugin-rest-endpoint-methods?dts";
+import type { EventPayloadMap } from "https://cdn.skypack.dev/@octokit/webhooks-types?dts";
+import "https://raw.githubusercontent.com/GoogleChrome/lighthouse-ci/v0.8.2/types/assert.d.ts";
+
+/* -- Setup -- */
+
+/** Github token for Authentication */
+const token = Deno.env.get("GITHUB_TOKEN");
+if (!token) throw new Error("Missing GITHUB_TOKEN");
+
+/** Path for workflow event */
+const path = Deno.env.get("GITHUB_EVENT_PATH");
+if (!path) throw new Error("Missing GITHUB_EVENT_PATH");
+
+/**
+ * https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads
+ */
+const payload: EventPayloadMap["push" | "pull_request"] = JSON.parse(
+	Deno.readTextFileSync(path)
+);
+
+const isPullRequestEvent = (
+	payload: EventPayloadMap[keyof EventPayloadMap]
+): payload is EventPayloadMap["pull_request"] =>
+	//@ts-expect-error -- We’re actually checking the type
+	typeof payload?.pull_request?.number === "number";
+
+/**
+ * One of two values depending on the workflow trigger event
+ *
+ * - on a "pull_request" event: the current Issue / PR number
+ * on which to add or update the Lighthouse report
+ *
+ * - on a "push" event: the original issue to track the latest report:
+ * https://github.com/guardian/dotcom-rendering/issues/4584
+ */
+const issue_number = isPullRequestEvent(payload)
+	? // If PullRequestEvent
+	  payload.pull_request.number
+	: // If PushEvent
+	  4584;
+
+console.log(`Using issue #${issue_number}`);
+
+console.log({ payload });
+
+/** The Lighthouse results directory  */
+const dir = "dotcom-rendering/.lighthouseci";
+
+const links: Record<string, string> = JSON.parse(
+	Deno.readTextFileSync(`${dir}/links.json`)
+);
+
+/** https://github.com/GoogleChrome/lighthouse-ci/blob/5963dcce0e88b8d3aedaba56a93ec4b93cf073a1/packages/utils/src/assertions.js#L15-L30 */
+interface AssertionResult {
+	url: string;
+	name:
+		| keyof Omit<LHCI.AssertCommand.AssertionOptions, "aggregationMethod">
+		| "auditRan";
+	operator: string;
+	expected: number;
+	actual: number;
+	values: number[];
+	passed: boolean;
+	level?: LHCI.AssertCommand.AssertionFailureLevel;
+	auditId?: string;
+	auditProperty?: string;
+	auditTitle?: string;
+	auditDocumentationLink?: string;
+	message?: string;
+}
+const results: AssertionResult[] = JSON.parse(
+	Deno.readTextFileSync(`${dir}/assertion-results.json`)
+);
+
+/* -- Definitions -- */
+
+/** The string to search for when looking for a comment */
+const REPORT_TITLE = "⚡️ Lighthouse report";
+const GIHUB_PARAMS = {
+	owner: "guardian",
+	repo: "dotcom-rendering",
+	issue_number,
+} as const;
+
+const octokit = new Octokit({ auth: token }) as {
```

Comment:
what is happening after the `as` ?

Reply:
I’ve refactored a bit to make it clearer that we’re building the type ourselves.